src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const cloneState = { ...state };
+  const resultArray = [];
+
+  actions.forEach(action => {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(cloneState, action.extraData);
```

Comment:
It's better to use spread operator for these operations.

Reply:
@bohdan-mykhailenko not necessarily. The spread operator and Object.assign() are very similar. The main difference is that spreading defines new properties, while Object.assign() sets them.

https://2ality.com/2016/10/rest-spread-properties.html#spread-defines-properties-objectassign-sets-them